python/foglamp/plugins/south/systeminfo/systeminfo.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Module for System Info async plugin """
+
+import time
+import asyncio
+import copy
+import uuid
+import datetime
+import sys
+import subprocess
+import socket
+import fcntl
+import struct
+import array
+
+from foglamp.services.core.connect import *
+from foglamp.common import logger
+from foglamp.plugins.common import utils
+from foglamp.services.south import exceptions
+from foglamp.services.south.ingest import Ingest
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2018 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_DEFAULT_CONFIG = {
+    'plugin': {
+        'description': 'System info async plugin',
+        'type': 'string',
+        'default': 'systeminfo'
+    },
+    'assetCode': {
+        'description': 'Asset Code',
+        'type': 'string',
+        'default': "system"
+    },
+    'sleepInterval': {
+        'description': 'Sleep Interval, in seconds, between two System info gathering',
+        'type': 'integer',
+        'default': "30"
+    },
+    'networkSnifferPeriod': {
+        'description': 'Interval, in seconds, for which network traffic is measured',
+        'type': 'integer',
+        'default': "2"
+    }
+}
+_LOGGER = logger.setup(__name__, level=20)
+
+
+def plugin_info():
+    """ Returns information about the plugin.
+    Args:
+    Returns:
+        dict: plugin information
+    Raises:
+    """
+
+    return {
+        'name': 'System Info plugin',
+        'version': '1.0',
+        'mode': 'async',
+        'type': 'south',
+        'interface': '1.0',
+        'config': _DEFAULT_CONFIG
+    }
+
+
+def plugin_init(config):
+    """ Initialise the plugin.
+    Args:
+        config: JSON configuration document for the South device configuration category
+    Returns:
+        data: JSON object to be used in future calls to the plugin
+    Raises:
+    """
+    data = copy.deepcopy(config)
+    return data
+
+
+def plugin_start(handle):
```

Comment:
why not its a poll plugin? does not this poll info from the system its installed on?!

Reply:
as discussed in last scrum, it is ok if this plugin stays async.